### PR TITLE
Clarify spec for constant of shape with dim_n = 0

### PR DESCRIPTION
**Pull request by neginraoof**
Both Numpy and PyTorch do support constructing a tensor of shape zero.
This PR clarifies the spec for ONNX constantofshape.

### CODE REVIEW

File: docs/Changelog.md [hunk 1]
```diff
@@ -8184,7 +8184,7 @@ This version of the operator has been available since version 9 of the default O
 
 <dl>
 <dt><tt>input</tt> : T1</dt>
-<dd>1D tensor. The shape of the expected output tensor. If empty tensor is given, the output would be a scalar.</dd>
+<dd>1D tensor. The shape of the expected output tensor. If empty tensor is given, the output would be a scalar. All values must be >= 0.</dd>
```
Comment (wschin) [inline docs/Changelog.md]: Must be `>0`? I believe the `=` case corresponds to a scalar and we already have `empty tensor` for that.

Comment (neginraoof) [reply docs/Changelog.md]: Yes, but we can a multidimensional tensor with one dim = 0.
Like:
numpy.full((2, 0), 4)
numpy.random.randn(2, 0)
or
torch.randn([0, 2, 3])

**Comment by neginraoof** [comment]
@wschin 
An example model that needs this is a torchvision model.
In case of no detection, the model creates an empty tensor of size (0, B, ...)

File: docs/Changelog.md (hunk 1 above)
Comment (gramalingam) [reply docs/Changelog.md]: Just to clarify: the output is expected to be an empty tensor, so the constant-value used is immaterial. Is that right?

Comment (neginraoof) [reply docs/Changelog.md]: Yes, that's correct.

**Comment by neginraoof** [comment]
cc @postrational @houseroad. Could you please help review this PR? Thanks.

**Pull request merged**